pandas/core/indexes/base.py

```diff
@@ -313,6 +313,11 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
             else:
                 return result
 
+        elif is_period_dtype(data) and not is_object_dtype(dtype):
```

Comment:
why the `and not is_object_dtype` ?

Reply:
We require that `Index(..., dtype=object)` always return an `Index`, and not a subclass. So this makes `Index(PeriodArray, dtype='object')` and `Index[ndarray[object]]`